### PR TITLE
Added a way to set a TTL to key values.

### PR DESCRIPTION
**Pull request by fulmicoton**
After the GC grace period, the key value will be subject to the GC just like regularly deleted KVs.

### CODE REVIEW

File: chitchat/src/state.rs [hunk 1]
```diff
@@ -1051,12 +1076,17 @@ mod tests {
...
         node_state.set("key", "1");
-        node_state.mark_for_deletion("key");
+        node_state.delete("key");
+        assert!(node_state.get("key").is_none());
         {
             let versioned_value = node_state.get_versioned("key").unwrap();
             assert_eq!(&versioned_value.value, "");
             assert_eq!(versioned_value.version, 2u64);
-            assert!(&versioned_value.is_tombstone());
+            assert!(&versioned_value.is_deleted());
```
Comment (trinity-1686a) [inline chitchat/src/state.rs:1085]:
```suggestion
            assert!(versioned_value.is_deleted());
```
i don't think the borrow is necessary, and i find it somewhat confusing (is it supposed to be a borrow, or a mistyped !)

File: chitchat/src/types.rs [hunk 2]
```diff
@@ -108,7 +133,7 @@ impl PartialEq for VersionedValue {
     fn eq(&self, other: &Self) -> bool {
         self.value.eq(&other.value)
             && self.version.eq(&other.version)
-            && self.is_tombstone().eq(&other.is_tombstone())
+            && self.is_deleted().eq(&other.is_deleted())
```
Comment (trinity-1686a) [inline chitchat/src/types.rs:136]: I'm not sure is_deleted is enough to qualify equality. `DeletionStatus::Set` and `DeletionStatus::DeleteAfterTtl` are considered equal by this statement. If that is intended, a comment should explain why

**Review by trinity-1686a — COMMENTED** [review]
looks sensible; see comments

File: chitchat/src/state.rs (hunk 1 above)
Comment (fulmicoton) [reply chitchat/src/state.rs:1085]: yes!

File: chitchat/src/types.rs (hunk 2 above)
Comment (fulmicoton) [reply chitchat/src/types.rs]: Thx. I switched to checking the discriminant, added a comment, and made PartialEq cfg test only.

**Pull request merged**